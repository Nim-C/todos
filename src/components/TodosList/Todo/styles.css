.trash-icon {
  width: 15px;
  color: #cfd2d4;
}

.trash-icon :hover {
  color: #dc3545;
}

.todo-name {
  font-family: 'Inter', sans-serif;
  color: #152536;
}

.todo-completed {
  text-decoration: line-through;
  color: #6b7176;
}

.custom-checkbox .custom-control-input:checked ~ .custom-control-label::before {
  background-color: var(--green);
  border: none;
}
.custom-checkbox .custom-control-input:focus ~ .custom-control-label::before {
  box-shadow: 0 0 3px var(--green);
  box-shadow: none;
  border-color: #adb5bd;
}
.custom-checkbox .custom-control-input:active ~ .custom-control-label::before {
  box-shadow: 0 0 3px var(--green);
  border-color: #adb5bd;
}
.custom-checkbox .custom-control-label::before {
  border-radius: 0.25rem;
}
.custom-control-label::before {
  background-color: #fff;
}
