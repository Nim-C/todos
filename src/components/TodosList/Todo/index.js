import React, { useCallback } from 'react'
import classnames from 'classnames'
import './styles.css'

import { ReactComponent as TrashCanIcon } from '../../../trash-solid.svg'
import { useDispatch } from 'react-redux'
import { removeTodo, toggleTodo } from '../../../features/todos/todosSlice'

const Todo = ({ id, name, completed }) => {
  const dispatch = useDispatch()

  const handleRemoveTodo = useCallback(() => {
    dispatch(removeTodo({ id }))
  }, [dispatch, id])

  const handleToggleTodo = useCallback(() => {
    dispatch(toggleTodo({ id }))
  }, [dispatch, id])

  return (
    <div
      className={classnames('card', 'rounded', 'bg-white', 'mb-4', 'col-12', {
        shadow: !completed,
        'shadow-sm': completed
      })}
    >
      <div
        className={classnames(
          'card-body',
          'd-flex',
          'justify-content-between',
          'todo-name'
        )}
      >
        <div className={classnames('custom-control', 'custom-checkbox')}>
          <input
            type="checkbox"
            className="custom-control-input"
            id={`todo-${id}-checkbox`}
            onChange={handleToggleTodo}
          />
          <label
            className={classnames('custom-control-label', {
              'todo-completed': completed
            })}
            htmlFor={`todo-${id}-checkbox`}
          >
            {name}
          </label>
        </div>
        <TrashCanIcon className="trash-icon" onClick={handleRemoveTodo} />
      </div>
    </div>
  )
}

export default Todo
