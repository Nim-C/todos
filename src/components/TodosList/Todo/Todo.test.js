import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import store from '../../../store'
import Todo from './'

test('renders single todo without failing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Todo id="013513515122" name="Dummy" completed="false" />
      </Provider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
