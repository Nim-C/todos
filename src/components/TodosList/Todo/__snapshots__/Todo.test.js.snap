// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders single todo without failing 1`] = `
<div
  className="card rounded bg-white mb-4 col-12 shadow-sm"
>
  <div
    className="card-body d-flex justify-content-between todo-name"
  >
    <div
      className="custom-control custom-checkbox"
    >
      <input
        className="custom-control-input"
        id="todo-013513515122-checkbox"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        className="custom-control-label todo-completed"
        htmlFor="todo-013513515122-checkbox"
      >
        Dummy
      </label>
    </div>
    <svg
      className="trash-icon"
      onClick={[Function]}
    >
      trash-solid.svg
    </svg>
  </div>
</div>
`;
